Objects/unicodeobject.c

```diff
@@ -2038,10 +2110,11 @@ PyUnicode_FromWideChar(const wchar_t *u, Py_ssize_t size)
     if (size == 0)
         _Py_RETURN_UNICODE_EMPTY();
 
-    /* Single character Unicode objects in the Latin-1 range are
-       shared when using this constructor */
-    if (size == 1 && (Py_UCS4)*u < 256)
-        return get_latin1_char((unsigned char)*u);
+    /* Single character Unicode objects can be shared when using this
+       constructor. */
+    if (size == 1) {
+        return unicode_char((Py_UCS4)*u);
```

Comment:
```
    def test_issue17223(self):
        # this used to crash
        if sizeof_wchar == 4:
            # U+FFFFFFFF is an invalid code point in Unicode 6.0
            invalid_str = b'\xff\xff\xff\xff'
        else:
            # PyUnicode_FromUnicode() cannot fail with 16-bit wchar_t
            self.skipTest("specific to 32-bit wchar_t")
        a = array.array('u', invalid_str)
        self.assertRaises(ValueError, a.tounicode)
        self.assertRaises(ValueError, str, a)
```

PyUnicode_FromWideChar() should raise ValueError when u is out of unicode range.

Reply:
Does not it do this? `unicode_char()` is an implementation of Python's `chr()`.

Reply:
`    assert(ch <= MAX_UNICODE);` is in top of `unicode_char()`

Reply:
Good catch!